pkg/validatorfactory/schema_patcher.go

```diff
@@ -0,0 +1,141 @@
+package validatorfactory
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"sigs.k8s.io/kubectl-validate/pkg/utils"
+)
+
+type SchemaPatch struct {
+	Slug        string
+	Description string
+
+	// (Inclusive) version range for which this patch applies
+	MinMinorVersion int
+	MaxMinorVersion int
+
+	// Nil is wildcard
+	AppliesToGV func(schema.GroupVersion) bool
+	Transformer utils.SchemaVisitor
+}
+
+var zero int64 = int64(0)
+var schemaPatches []SchemaPatch = []SchemaPatch{
+	{
+		Slug:            "AllowEmptyByteFormat",
+		Description:     "Work around discrepency between treatment of native vs CRD `byte` strings. Native types allow empty, CRDs do not",
+		MinMinorVersion: 0,
+		MaxMinorVersion: 0,
+		AppliesToGV:     nil,
+		Transformer: utils.PostorderVisitor(func(ctx utils.VisitingContext, s *spec.Schema) bool {
+			if s.Format != "byte" || len(s.Type) != 1 || s.Type[0] != "string" {
+				return true
+			}
+
+			// Change format to "", and add new `$and: {$or: [{format: "byte"}, {maxLength: 0}]}
+			s.AllOf = append(s.AllOf, spec.Schema{
+				SchemaProps: spec.SchemaProps{
+					AnyOf: []spec.Schema{
+						{
+							SchemaProps: spec.SchemaProps{
+								Format: s.Format,
+							},
+						},
+						{
+							SchemaProps: spec.SchemaProps{
+								MaxLength: &zero,
+							},
+						},
+					},
+				},
+			})
+			s.Format = ""
+			return true
+		}),
+	},
+	{
+		Slug:        "AllowEmptyDateTime",
```

Comment:
This also can't run against CRDs

Reply:
How about `metadata.creationTimestamp` ?

Reply:
In the published schema for CRDs, `metadata.creationTimestamp` is usually specified as a `$ref` to the internal `k8s` type. 

There should be a way from the name of the schema definition ref to tell that it is native to k8s (e.g. `io.k8s.apimachinery.pkg.apis.meta.v1.Time` with `io.k8s` prefix).

Reply:
But yes, this should run against creationTImestamp since it is deserialized and validated as a native type rather than against the schema in k8s

Reply:
Unfortunately this won't work for aggregated builtin types. That can't be helped until we have a server-side way of discerning CRDs vs native types in the schema. But by the time that happens this bug will be fixed

Reply:
Oh yeah, aggregated builtin types 🙈